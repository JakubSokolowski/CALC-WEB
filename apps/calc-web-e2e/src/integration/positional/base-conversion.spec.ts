import {
    getBconvConvertButton,
    getComplementInput, getConversionResult,
    getDecimalValueInput,
    getInputBaseInput,
    getInputStrInput,
    getIntegralConversionGrid,
    getOutputBaseInput,
    getShowComplementSwitchOff,
    getShowComplementSwitchOn,
    getShowDecimalValueSwitchOff,
    getShowDecimalValueSwitchOn
} from '../../support/bconv';

describe('Base converter error labels', () => {
    beforeEach(() => {
        cy.fixCypressSpec(__filename);
        cy.clearLocalStorage();
        cy.visit('#/tools/positional/base-converter');
    });

    it('should display error when input base is invalid and disable convert button', () => {
        const errorText = 'Base must be between 2 and 99';
        const labelId = '#inputBase-helper-text';

        getInputBaseInput().type('1');
        cy.get(labelId).contains(errorText);
        getBconvConvertButton().should('be.disabled');

        getInputBaseInput().clear().type('200');
        cy.get(labelId).contains(errorText);
        getBconvConvertButton().should('be.disabled');

        getInputBaseInput().clear().type('10');
        cy.get(labelId).should('not.exist');
    });

    it('should display error when output base is invalid and disable convert button', () => {
        const errorText = 'Base must be between 2 and 99';
        const labelId = '#outputBase-helper-text';
        getOutputBaseInput().clear().type('1');
        cy.get(labelId).contains(errorText);
        getBconvConvertButton().should('be.disabled');

        getOutputBaseInput().clear().type('200');
        cy.get(labelId).contains(errorText);
        getBconvConvertButton().should('be.disabled');

        getOutputBaseInput().clear().type('10');
        cy.get(labelId).should('not.exist');
    });

    it('should display error when representation does not match the base', () => {
        const base = 10;
        const errorText = `Representation strings contains invalid digits for base ${base}`;
        const labelId = '#inputStr-helper-text';

        getInputBaseInput().clear().type(`${base}`);
        getInputStrInput().clear().type('FFA');

        cy.get(labelId).contains(errorText);
    });
});

describe('Base converter show decimal value/complement switches', () => {
    beforeEach(() => {
        cy.clearLocalStorage();
        cy.visit('#/tools/positional/base-converter');
    });

    it('should show/hide input decimal value on switch change', () => {
        // should show proper value
        getShowDecimalValueSwitchOn();
        cy.baseConverterInput('-11011', 2, 16);
        getDecimalValueInput().should('have.value', '-27');

        // should hide it on switch change
        getShowDecimalValueSwitchOn().click();
        getDecimalValueInput().should('not.exist');

        // switch state (off) should persist on reload
        cy.reload();
        getShowDecimalValueSwitchOff();
    });

    it('should show/hide input complement on switch change', () => {
        // should show proper value
        getShowComplementSwitchOn();
        cy.baseConverterInput('-11011', 2, 16);
        getComplementInput().should('have.value', '(1)00101');

        // should hide it on switch change
        getShowComplementSwitchOn().click();
        getComplementInput().should('not.exist');

        // switch state (off) should persist on reload
        cy.reload();
        getShowComplementSwitchOff();
    });
});

describe('Base converter conversion', () => {
    beforeEach(() => {
        cy.clearLocalStorage();
        cy.visit('#/tools/positional/base-converter');
    });

    it('should convert to another base when number has no fraction part', () => {
        const inputStr = '123';
        const inputBase = 10;
        const outputBase = 2;

        cy.baseConverterInput(inputStr, inputBase, outputBase);
        getConversionResult().toMatchSnapshot();
        getIntegralConversionGrid().toMatchSnapshot({});
    });
});
