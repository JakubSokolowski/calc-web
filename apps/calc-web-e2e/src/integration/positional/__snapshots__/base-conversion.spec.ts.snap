exports[`Base converter conversion > should convert to another base when number has no fraction part #0`] = `
<div data-test="bconv-result-1111011">
  <div>
    <div class="ConversionResult-equation">
      <div class="css-18uwm9r">
        <div data-test="conversion-to-decimal" class="ConversionToDecimal-row">
          <div class="" data-mui-internal-clone-element="true">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow
                        ><mn>12</mn><msub><mn>3</mn><mn>10</mn></msub></mrow
                      ><annotation encoding="application/x-tex"
                        >123_{10}</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.79444em;vertical-align:-0.15em;"
                    ></span
                    ><span class="mord">1</span><span class="mord">2</span
                    ><span class="mord"
                      ><span class="mord">3</span
                      ><span class="msupsub"
                        ><span class="vlist-t vlist-t2"
                          ><span class="vlist-r"
                            ><span
                              class="vlist"
                              style="height:0.30110799999999993em;"
                              ><span
                                style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"
                                ><span
                                  class="pstrut"
                                  style="height:2.7em;"
                                ></span
                                ><span class="sizing reset-size6 size3 mtight"
                                  ><span class="mord mtight"
                                    ><span class="mord mtight">1</span
                                    ><span class="mord mtight">0</span></span
                                  ></span
                                ></span
                              ></span
                            ><span class="vlist-s">​</span></span
                          ><span class="vlist-r"
                            ><span class="vlist" style="height:0.15em;"
                              ><span></span></span></span></span></span></span></span></span></span
            ></span>
          </div>
          <div class="ConversionToDecimal-symbol">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow><mo>=</mo></mrow
                      ><annotation encoding="application/x-tex"
                        >=</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.36687em;vertical-align:0em;"
                    ></span
                    ><span class="mrel">=</span></span
                  ></span
                ></span
              ></span
            >
          </div>
          <span
            ><span class="katex"
              ><span class="katex-mathml"
                ><math xmlns="http://www.w3.org/1998/Math/MathML"
                  ><semantics
                    ><mrow
                      ><mo stretchy="false">(</mo><mn>1</mn><mo>∗</mo><mn>1</mn
                      ><msup><mn>0</mn><mn>2</mn></msup
                      ><mo stretchy="false">)</mo><mo>+</mo></mrow
                    ><annotation encoding="application/x-tex"
                      >(1 * 10^{2}) +
                    </annotation></semantics
                  ></math
                ></span
              ><span class="katex-html" aria-hidden="true"
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mopen">(</span><span class="mord">1</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span
                  ><span class="mbin">∗</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span></span
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1.064108em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mord">1</span
                  ><span class="mord"
                    ><span class="mord">0</span
                    ><span class="msupsub"
                      ><span class="vlist-t"
                        ><span class="vlist-r"
                          ><span
                            class="vlist"
                            style="height:0.8141079999999999em;"
                            ><span style="top:-3.063em;margin-right:0.05em;"
                              ><span class="pstrut" style="height:2.7em;"></span
                              ><span class="sizing reset-size6 size3 mtight"
                                ><span class="mord mtight"
                                  ><span class="mord mtight">2</span></span
                                ></span
                              ></span
                            ></span
                          ></span
                        ></span
                      ></span
                    ></span
                  ><span class="mclose">)</span
                  ><span class="mord">+</span></span
                ></span
              ></span
            ></span
          ><span
            ><span class="katex"
              ><span class="katex-mathml"
                ><math xmlns="http://www.w3.org/1998/Math/MathML"
                  ><semantics
                    ><mrow
                      ><mo stretchy="false">(</mo><mn>2</mn><mo>∗</mo><mn>1</mn
                      ><msup><mn>0</mn><mn>1</mn></msup
                      ><mo stretchy="false">)</mo><mo>+</mo></mrow
                    ><annotation encoding="application/x-tex"
                      >(2 * 10^{1}) +
                    </annotation></semantics
                  ></math
                ></span
              ><span class="katex-html" aria-hidden="true"
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mopen">(</span><span class="mord">2</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span
                  ><span class="mbin">∗</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span></span
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1.064108em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mord">1</span
                  ><span class="mord"
                    ><span class="mord">0</span
                    ><span class="msupsub"
                      ><span class="vlist-t"
                        ><span class="vlist-r"
                          ><span
                            class="vlist"
                            style="height:0.8141079999999999em;"
                            ><span style="top:-3.063em;margin-right:0.05em;"
                              ><span class="pstrut" style="height:2.7em;"></span
                              ><span class="sizing reset-size6 size3 mtight"
                                ><span class="mord mtight"
                                  ><span class="mord mtight">1</span></span
                                ></span
                              ></span
                            ></span
                          ></span
                        ></span
                      ></span
                    ></span
                  ><span class="mclose">)</span
                  ><span class="mord">+</span></span
                ></span
              ></span
            ></span
          ><span
            ><span class="katex"
              ><span class="katex-mathml"
                ><math xmlns="http://www.w3.org/1998/Math/MathML"
                  ><semantics
                    ><mrow
                      ><mo stretchy="false">(</mo><mn>3</mn><mo>∗</mo><mn>1</mn
                      ><msup><mn>0</mn><mn>0</mn></msup
                      ><mo stretchy="false">)</mo></mrow
                    ><annotation encoding="application/x-tex"
                      >(3 * 10^{0})
                    </annotation></semantics
                  ></math
                ></span
              ><span class="katex-html" aria-hidden="true"
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mopen">(</span><span class="mord">3</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span
                  ><span class="mbin">∗</span
                  ><span
                    class="mspace"
                    style="margin-right:0.2222222222222222em;"
                  ></span></span
                ><span class="base"
                  ><span
                    class="strut"
                    style="height:1.064108em;vertical-align:-0.25em;"
                  ></span
                  ><span class="mord">1</span
                  ><span class="mord"
                    ><span class="mord">0</span
                    ><span class="msupsub"
                      ><span class="vlist-t"
                        ><span class="vlist-r"
                          ><span
                            class="vlist"
                            style="height:0.8141079999999999em;"
                            ><span style="top:-3.063em;margin-right:0.05em;"
                              ><span class="pstrut" style="height:2.7em;"></span
                              ><span class="sizing reset-size6 size3 mtight"
                                ><span class="mord mtight"
                                  ><span class="mord mtight">0</span></span
                                ></span
                              ></span
                            ></span
                          ></span
                        ></span
                      ></span
                    ></span
                  ><span class="mclose">)</span></span
                ></span
              ></span
            ></span
          >
          <div class="ConversionToDecimal-symbol">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow><mo>=</mo></mrow
                      ><annotation encoding="application/x-tex"
                        >=</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.36687em;vertical-align:0em;"
                    ></span
                    ><span class="mrel">=</span></span
                  ></span
                ></span
              ></span
            >
          </div>
          <div class="" data-mui-internal-clone-element="true">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow
                        ><mn>12</mn><msub><mn>3</mn><mn>10</mn></msub></mrow
                      ><annotation encoding="application/x-tex"
                        >123_{10}</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.79444em;vertical-align:-0.15em;"
                    ></span
                    ><span class="mord">1</span><span class="mord">2</span
                    ><span class="mord"
                      ><span class="mord">3</span
                      ><span class="msupsub"
                        ><span class="vlist-t vlist-t2"
                          ><span class="vlist-r"
                            ><span
                              class="vlist"
                              style="height:0.30110799999999993em;"
                              ><span
                                style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"
                                ><span
                                  class="pstrut"
                                  style="height:2.7em;"
                                ></span
                                ><span class="sizing reset-size6 size3 mtight"
                                  ><span class="mord mtight"
                                    ><span class="mord mtight">1</span
                                    ><span class="mord mtight">0</span></span
                                  ></span
                                ></span
                              ></span
                            ><span class="vlist-s">​</span></span
                          ><span class="vlist-r"
                            ><span class="vlist" style="height:0.15em;"
                              ><span></span></span></span></span></span></span></span></span></span
            ></span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div class="css-9qn66q">
        <div class="ResultEquation-row" data-test="result-equation">
          <div class="" data-mui-internal-clone-element="true">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow
                        ><mn>12</mn><msub><mn>3</mn><mn>10</mn></msub></mrow
                      ><annotation encoding="application/x-tex"
                        >123_{10}</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.79444em;vertical-align:-0.15em;"
                    ></span
                    ><span class="mord">1</span><span class="mord">2</span
                    ><span class="mord"
                      ><span class="mord">3</span
                      ><span class="msupsub"
                        ><span class="vlist-t vlist-t2"
                          ><span class="vlist-r"
                            ><span
                              class="vlist"
                              style="height:0.30110799999999993em;"
                              ><span
                                style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"
                                ><span
                                  class="pstrut"
                                  style="height:2.7em;"
                                ></span
                                ><span class="sizing reset-size6 size3 mtight"
                                  ><span class="mord mtight"
                                    ><span class="mord mtight">1</span
                                    ><span class="mord mtight">0</span></span
                                  ></span
                                ></span
                              ></span
                            ><span class="vlist-s">​</span></span
                          ><span class="vlist-r"
                            ><span class="vlist" style="height:0.15em;"
                              ><span></span></span></span></span></span></span></span></span></span
            ></span>
          </div>
          <div class="ResultEquation-symbol">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow><mo>=</mo></mrow
                      ><annotation encoding="application/x-tex"
                        >=</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.36687em;vertical-align:0em;"
                    ></span
                    ><span class="mrel">=</span></span
                  ></span
                ></span
              ></span
            >
          </div>
          <div class="" data-mui-internal-clone-element="true">
            <span
              ><span class="katex"
                ><span class="katex-mathml"
                  ><math xmlns="http://www.w3.org/1998/Math/MathML"
                    ><semantics
                      ><mrow
                        ><mn>111101</mn><msub><mn>1</mn><mn>2</mn></msub></mrow
                      ><annotation encoding="application/x-tex"
                        >1111011_{2}</annotation
                      ></semantics
                    ></math
                  ></span
                ><span class="katex-html" aria-hidden="true"
                  ><span class="base"
                    ><span
                      class="strut"
                      style="height:0.79444em;vertical-align:-0.15em;"
                    ></span
                    ><span class="mord">1</span><span class="mord">1</span
                    ><span class="mord">1</span><span class="mord">1</span
                    ><span class="mord">0</span><span class="mord">1</span
                    ><span class="mord"
                      ><span class="mord">1</span
                      ><span class="msupsub"
                        ><span class="vlist-t vlist-t2"
                          ><span class="vlist-r"
                            ><span
                              class="vlist"
                              style="height:0.30110799999999993em;"
                              ><span
                                style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"
                                ><span
                                  class="pstrut"
                                  style="height:2.7em;"
                                ></span
                                ><span class="sizing reset-size6 size3 mtight"
                                  ><span class="mord mtight"
                                    ><span class="mord mtight">2</span></span
                                  ></span
                                ></span
                              ></span
                            ><span class="vlist-s">​</span></span
                          ><span class="vlist-r"
                            ><span class="vlist" style="height:0.15em;"
                              ><span></span></span></span></span></span></span></span></span></span
            ></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Base converter conversion > should convert to another base when number has no fraction part #1`] = `
<div class="HoverGrid-cellContent" id="integral-conversion-grid">
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-0"
        class="GridCell-defaultCell"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-0"
        class="GridCell-defaultCell"
      >
        2
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-0"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        3
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-0"
        class="GridCell-defaultCell"
      >
        6
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-0"
        class="GridCell-defaultCell"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-0"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-0"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
  </div>
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-1"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-1"
        class="GridCell-defaultCell"
      >
        6
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-1"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-1"
        class="GridCell-defaultCell"
      >
        3
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-1"
        class="GridCell-defaultCell"
      >
        0
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-1"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-1"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
  </div>
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-2"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-2"
        class="GridCell-defaultCell"
      >
        3
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-2"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        0
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-2"
        class="GridCell-defaultCell"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-2"
        class="GridCell-defaultCell"
      >
        5
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-2"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-2"
        class="GridCell-highlightedCell"
      >
        0
      </div>
    </div>
  </div>
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-3"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-3"
        class="GridCell-defaultCell"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-3"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        5
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-3"
        class="GridCell-defaultCell"
      >
        7
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-3"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-3"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-3"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
  </div>
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-4"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-4"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-4"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        7
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-4"
        class="GridCell-defaultCell"
      >
        3
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-4"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-4"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-4"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
  </div>
  <div class="HoverGrid-gridRow">
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-0-5"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-1-5"
        class="GridCell-defaultCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-2-5"
        class="GridCell-defaultCell GridCell-verticalLine"
      >
        3
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-3-5"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-4-5"
        class="GridCell-highlightedCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-5-5"
        class="GridCell-highlightedCell"
      ></div>
    </div>
    <div class="css-r22d85">
      <div
        data-test="integral-conversion-grid-6-5"
        class="GridCell-highlightedCell"
      >
        1
      </div>
    </div>
  </div>
</div>
`;
